src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -363,13 +364,60 @@ public void run(final RunNotifier notifier) {
             statement.evaluate();
         } catch (AssumptionViolatedException e) {
             testNotifier.addFailedAssumption(e);
-        } catch (StoppedByUserException e) {
-            throw e;
         } catch (Throwable e) {
-            testNotifier.addFailure(e);
```

Comment:
I agree, this was one of my first attempt to fix this issue, but returning to my points in the previous conversation - if we will simply pass `MultipleFailureException` to notifier we will get behavior changes, therefore I decided to unpack all failures manually and notify listeners about all errors except `StoppedByUserException`.
Other solution can be add filtering inside `EachTestNotifier` - modify `addFailure()` method to skip `StoppedByUserException` and pass all other errors. This can make code even simpler, but again this will lead to behavior changes - such alteration will make JUnit to completely ignore `StoppedByUserException` thrown from the test method manually, which I believe is not acceptable.